File: CItem_proj_2/mask.py
from citem import CItem
from typing import List

class Mask(CItem):
    def __init__(self, name: str, properties: str, isOn: bool = False):
        super(Mask,self).__init__(name)
        self._properties = properties
        if self._inUse == False:
            self._isOn = False
        else:
            self._isOn = isOn

    @property
    def isOn(self) -> bool:
        return self._isOn

    def putOn(self) -> None:
        if self._inUse == True:
            self._isOn = True
    
    def takeOff(self) -> None:
        self._isOn = False

    @property
    def properties(self) -> str:
        return self._properties

    def stopUsing(self) -> None:
        self._inUse = False   
        self._isOn = False 

    def applyC(self, newDirect: str) -> None:
        if newDirect not in self.ACCEPTABLE_DIRECTIONS or self._acquired == False:
            self._cApplied = "none"
        else:
            self._cApplied = newDirect
        if self._cApplied == "none":
            self._inUse = False
            self._isOn = False

File: CItem_proj_2_cpp/src/Mask.cpp
#include "Mask.h"
#include <iostream>

namespace item {
    bool isOn;
    
    Mask::Mask(std::string _name, std::string _properties, bool _isOn) 
        : CItem(_name), properties(_properties), isOn(_isOn)
    {
        if(!getUsing()) isOn = false;
        else isOn = true;
    }

    bool Mask::getIsOn() const {
        return isOn;
    }

    void Mask::putOn() {
        if(getUsing()) isOn = true;
    }
    void Mask::takeOff() {
        isOn = false;
    }

    std::string Mask::getProperties() const {
        return properties;
    }

    void Mask::stopUsing() {
        CItem::stopUsing();
        isOn = false;
    }

    void Mask::setCDirection(std::string newDirect) {
        bool accept = false;
        for(int i = 0; i < 4; i++) {
            if(newDirect == ACCEPTABLE_DIRECTIONS[i]) accept = true;
        }

        if(!accept) CItem::setCDirection("none");
        else CItem::setCDirection(newDirect);
        if (getCDirection() == "none") stopUsing();
    }
}
